### PR TITLE
Adds Typescript and Tailwind

### PR DESCRIPTION
**Pull request by soupaJ**
This PR adds:
* Tailwind for handling css
* Typescript

I tried to styles it as close to the original. I checked the js and css files sizes after tailwind and posted the results. Here's what I found:

|Original|Now|
|-------|-----|
|||

#### Original

|Page  | Original   | Now      | 
|----------|--------------|-------|
|Homepage|| |
| Gist | |   |
|Gist w/ comments|||
|Unsupported|| |

### CODE REVIEW

File: components/link/link.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  variant?: "subtle" | "button" | "unstyled";
+}
+
+const variantClasses = {
+  normal: "text-sky-600 no-underline hover:underline",
+  subtle: "text-gray-400",
+  button:
+    "rounded-md px-3 py-2 text-sky-600 hover:bg-blue-100 dark:hover:bg-blue-900 dark:hover:text-sky-300/90",
+  unstyled: "",
+};
```
Comment (jkulton) [inline components/link/link.tsx:14]: Oohh, I like this. Nice abstraction.

**Review by jkulton — COMMENTED** [review]
This looks great @soupaJ, thank you! I left a few comments, mostly just a few design tweaks. Feel free to push back on them if you feel strongly though.

One other thing I meant to ask, does the addition of ts change the steps for running the app locally? (Other than `npm i` to grab the new stuff?)

**Review by jkulton — COMMENTED** [review]
Were block-level code blocks on the light theme intentionally switched over to dark backgrounds or was this just a result of prose/style changes?

Screen Shot 2022-03-21 at 8 07 45 PM

**Comment by soupaJ** [comment]
This was the result of using `prose` from tailwind. There's a couple of stylistic choices that are different. Another thing I noticed is that inline code snippets are wrapped like `` `inline code`  ``.

Example:


Let me know if you'd like to revert these.

**Comment by jkulton** [comment]
I slightly prefer lighter code blocks on light mode, but we can always restyle that later. I think the prose styles are fine for now!

**Comment by soupaJ** [comment]
> One other thing I meant to ask, does the addition of ts change the steps for running the app locally? (Other than npm i to grab the new stuff?)

No, it should run the same.

**Comment by jkulton** [comment]
Ah, GitHub Actions ci build process is asking us to add `@types/react` to dev deps:

```
> gistdoc@ build /home/runner/work/gistdoc/gistdoc
> next build

warn  - SWC minify release candidate enabled. https://nextjs.org/docs/messages/swc-minify-enabled
info  - Checking validity of types...
It looks like you're trying to use TypeScript but do not have the required package(s) installed.

Please install @types/react by running:

	npm install --save-dev @types/react

If you are not trying to use TypeScript, please remove the tsconfig.json file from your package root (and any TypeScript files in your pages directory).

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gistdoc@ build: `next build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the gistdoc@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2022-03-22T02_2[7](https://github.com/jkulton/gistdoc/runs/5637725060?check_suite_focus=true#step:6:7)_44_405Z-debug.log
Error: Process completed with exit code 1.
```

**Review by jkulton — APPROVED** [review]
Great work!

**Pull request merged**